api/python/t4/data_transfer.py

```diff
@@ -126,44 +126,61 @@ def _upload_file(callback, size, src_path, dest_bucket, dest_key, override_meta)
                 Key=dest_key,
                 Metadata={HELIUM_METADATA: json.dumps(meta)},
             )
-            callback(fd.tell())
+            ctx.progress(fd.tell())
 
         version_id = resp.get('VersionId')  # Absent in unversioned buckets.
-        return make_s3_url(dest_bucket, dest_key, version_id)
+        ctx.done(make_s3_url(dest_bucket, dest_key, version_id))
     else:
         resp = s3_client.create_multipart_upload(
             Bucket=dest_bucket,
             Key=dest_key,
             Metadata={HELIUM_METADATA: json.dumps(meta)},
         )
         upload_id = resp['UploadId']
-        parts = []
-        with open(src_path, 'rb') as fd:
-            for i in itertools.count(1):
-                # TODO(dima): Better progress callback.
-                chunk = fd.read(s3_transfer_config.multipart_chunksize)
-                if not chunk:
-                    break
-                part = s3_client.upload_part(
-                    Body=chunk,
+
+        chunk_offsets = list(range(0, size, s3_transfer_config.multipart_chunksize))
```

Comment:
Can you write a test where the chucksize is very small to verify it produces enough and correct upload parts?